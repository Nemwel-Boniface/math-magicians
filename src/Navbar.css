.navBar {
  height: 120px;
}

.navBar nav {
  display: flex;
  justify-content: space-between;
}

.navBar nav h1 {
  font-size: 45px;
}

.navBar nav ul {
  display: flex;
  margin: auto 0;
}

.navBar nav ul li {
  list-style-type: none;
  margin-right: 20px;
}

.bar {
  padding-right: 10px;
  border-right: 2px solid #000;
}
