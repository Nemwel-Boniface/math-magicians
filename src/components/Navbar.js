import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div data-testid="Navbar" className="navBar">
    <nav>
      <h1>Math Magicians</h1>
      <ul>
        <li className="bar"><Link data-testid="home-link" to="/">Home</Link></li>
        <li className="bar"><Link data-testid="calculator-link" to="/calculator">Calculator</Link></li>
        <li><Link data-testid="quote-link" to="/quote">Quote</Link></li>
      </ul>
    </nav>
  </div>
);

export default Navbar;
