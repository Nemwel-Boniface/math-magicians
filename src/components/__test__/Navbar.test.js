
import { render, screen, fireEvent } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import '@testing-library/jest-dom';
import Navbar from '../Navbar';

expect.extend({ toMatchDiffSnapshot });

it('Check if the First click is the Home page', () => {
  const { asFragment } = render(<HashRouter><Navbar /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Home/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

it('Check if the Second click is the Calculator page', () => {
  const { asFragment } = render(<HashRouter><Navbar /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Calculator/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

it('Check if the third click is the Quotes page', () => {
  const { asFragment } = render(<HashRouter><Navbar /></HashRouter>);
  const firstRender = asFragment();
  fireEvent.click(screen.getByText(/Quote/));

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});

