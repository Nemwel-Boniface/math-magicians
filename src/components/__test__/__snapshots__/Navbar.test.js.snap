// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the First click is the Home page 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Check if the Second click is the Calculator page 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Check if the third click is the Quotes page 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Make sure events does not make differences in snapshot (Nav Component) | [Home] Menu Click  Event Trigger 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
